### PR TITLE
Introducing vector image for robot display, and more

### PR DESCRIPTION
**Pull request by rospogrigio**
Introduced a vectorial drawing of the robot (see image). It is now possible to select both fill color and outline. It is scalable, for best results the robot size should be at least 16 pixels, so the map should be scaled accordingly. I initially tried using an image and rotating it, but the result is not so good so I moved to vectorial drawing.

Moreover, introduced the orientation of the charger, with the possibility to both fill color and outline as well. Marker turned into a semicircle, in order to show the orientation.

image

### CODE REVIEW

**Comment by PiotrMachowski** [comment]
What scale have you used?

File: custom_components/xiaomi_cloud_map_extractor/common/image_handler.py [hunk 1]
```diff
@@ -181,6 +185,57 @@ def draw_texts(image: ImageData, texts):
     def draw_layer(image: ImageData, layer_name):
         ImageHandler.__draw_layer__(image, image.additional_layers[layer_name])
 
+    @staticmethod
+    def __draw_image__(image: ImageData, center, r, outline, fill):
```
Comment (PiotrMachowski) [inline custom_components/xiaomi_cloud_map_extractor/common/image_handler.py:189]: This method is unnecessary

File: custom_components/xiaomi_cloud_map_extractor/common/image_handler.py [hunk 2]
```diff
@@ -209,7 +274,7 @@ def draw_func(draw: ImageDraw):
             s = path.path[0].to_img(image.dimensions)
             for point in path.path[1:]:
                 e = point.to_img(image.dimensions)
-                draw.line([s.x * scale, s.y * scale, e.x * scale, e.y * scale], width=int(scale), fill=color)
+                draw.line([s.x * scale, s.y * scale, e.x * scale, e.y * scale], width=int(scale)*2, fill=color)
```
Comment (PiotrMachowski) [inline custom_components/xiaomi_cloud_map_extractor/common/image_handler.py:277]: Why?

**Comment by PiotrMachowski** [comment]
It looks really nice 👍

**Comment by rospogrigio** [comment]
> What scale have you used?

4x.

File: custom_components/xiaomi_cloud_map_extractor/common/image_handler.py (hunk 1 above)
Comment (rospogrigio) [reply custom_components/xiaomi_cloud_map_extractor/common/image_handler.py:189]: Ah yes, I started developing it when I was trying to use an image but then dropped the idea and forgot to remove it.

File: custom_components/xiaomi_cloud_map_extractor/common/image_handler.py (hunk 2 above)
Comment (rospogrigio) [reply custom_components/xiaomi_cloud_map_extractor/common/image_handler.py:277]: Ooops, this is just a variation, I prefer the path to be a little more visible. Just my preference, we can remove it or make it configurable as well.

Comment (PiotrMachowski) [reply custom_components/xiaomi_cloud_map_extractor/common/image_handler.py:277]: It shouldn't be a problem to make it configurable 👍

**Comment by PiotrMachowski** [comment]
> > What scale have you used?
> 
> 4x.

I might cause some performance issues on some devices (rpi3), 2x seems to be a value that can be handled by all platforms

**Comment by rospogrigio** [comment]
> I might cause performance issues on some devices (rpi3), 2x seems to be a value that can be handled by all platforms

Well, then it gets a bit uglier but still does its job:

image

**Comment by PiotrMachowski** [comment]
There were some issues with setups that used larger scale: https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues/86

Ultimately I want to change a library used for image rendering to use something lighter than pillow (but I haven't found anything interesting so far). There already was a concept of using using cairo instead (https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues/102), I might have to reconsider this library

**Comment by PiotrMachowski** [comment]
By the way, do not worry about failing check, it has been fixed in a different branch

**Comment by rospogrigio** [comment]
OK, I removed the unused method and restored the path width, if you'd like to merge this.
I am publishing another PR with some optimizations 😉

**Comment by PiotrMachowski** [comment]
I will merge it to `dev` branch and release it alongside support for Dreame vacuums

**Pull request merged**